src/parser/content-evaluator-module.ts

```diff
@@ -0,0 +1,141 @@
+import Decimal from "decimal.js";
+import { encodingForModel } from "js-tiktoken";
+import OpenAI from "openai";
+import configuration from "../configuration/config-reader";
+import { GetActivity } from "../get-activity";
+import { GitHubIssue } from "../github-types";
+import { GithubComment, Module, Result } from "./processor";
+
+/**
+ * Evaluates and rates comments.
+ */
+export class ContentEvaluatorModule implements Module {
+  readonly _openAi = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });
+  readonly configuration = configuration.contentEvaluator;
+
+  get enabled(): boolean {
+    return this.configuration.enabled;
+  }
+
+  async transform(data: Readonly<GetActivity>, result: Result) {
+    const promises: Promise<void>[] = [];
+
+    for (const key of Object.keys(result)) {
+      const currentElement = result[key];
+      const comments = currentElement.comments || [];
+      const specificationBody = (data.self as GitHubIssue)?.body;
+
+      if (specificationBody && comments.length) {
+        promises.push(
+          this._processComment(
+            comments,
```

Comment:
`_processComment` will modify the comment instead of returning the comments which I think can be confusing. I saw this approach in this code multiple times for example Modules modify the result instead of returning a fresh result. While this is not wrong I'm not really a fan of functions modifying parameters because it can lead to bugs and it's not clear to the user of the function

Reply:
I named the function `transform` on purpose and also marked `data` to be `ReadOnly` to explicitly note that only the `result` gets transformed. Since everything happens inside the class itself I thought it would not be so prone for accidents, will change it then